frontend/app/src/components/NavigationMenu.vue

```diff
@@ -50,11 +55,11 @@ import { Component, Vue, Prop } from 'vue-property-decorator';
 import NavigationMenuItem from '@/components/NavigationMenuItem.vue';
 
 type NavItem = {
-  readonly type: 'item' | 'group';
-  readonly text: string;
-  readonly route: string;
+  readonly type: 'item' | 'group' | 'divider';
+  readonly text?: string;
```

Comment:
Wouldn't it be better instead of making everything optional to make a `DividerItem` that has `type: 'divider'` and then make the array type `GroupNavItem | Divider`?

Reply:
This way if we don't have a divider the tsc will complain about missing properties :)

Reply:
Not sure I'm convinced but since you asked nicely... ;) I also split out the GroupItem into a separate type where `items` aren't optional, since we're taking the avenue of making sure that the type strictly adheres to the expected content.

Reply:
You are the best, thank you :smile: